src/Vulkan/VMASharp_API_For_Review.md

```diff
@@ -0,0 +1,307 @@
+    public override IntPtr Map();
+    public override void Unmap();
+  }
+  public class MapMemoryException : VulkanResultException
+  {
+    public MapMemoryException(string message);
+    public MapMemoryException(Result res);
+    public MapMemoryException(string message, Result res);
+  }
+  public enum MemoryUsage
+  {
+    Unknown,
+    GPU_Only,
+    CPU_Only,
+    CPU_To_GPU,
+    GPU_To_CPU,
+    CPU_Copy,
+    GPU_LazilyAllocated,
+  }
+  [Flags]
+  public enum PoolCreateFlags
+  {
+    IgnoreBufferImageGranularity = 1,
+    LinearAlgorithm = 16, // 0x00000010
+    BuddyAlgorithm = 32, // 0x00000020
+  }
+  public struct PoolStats
+  {
+    public long Size;
+    public long UnusedSize;
+    public int AllocationCount;
+    public int UnusedRangeCount;
+    public long UnusedRangeSizeMax;
+    public int BlockCount;
+  }
+  public struct StatInfo
+  {
+    public int BlockCount;
+    public int AllocationCount;
+    public int UnusedRangeCount;
+    public long UsedBytes;
+    public long UnusedBytes;
+    public long AllocationSizeMin;
+    public long AllocationSizeAvg;
+    public long AllocationSizeMax;
+    public long UnusedRangeSizeMin;
+    public long UnusedRangeSizeAvg;
+    public long UnusedRangeSizeMax;
+  }
+  public class Stats
+  {
+    public readonly StatInfo[] MemoryType;
+    public readonly StatInfo[] MemoryHeap;
+    public StatInfo Total;
+  }
+  public enum SuballocationType
+  {
+    Free,
+    Unknown,
+    Buffer,
+    Image_Unknown,
+    Image_Linear,
+    Image_Optimal,
+  }
+  public class ValidationFailedException : ApplicationException
+  {
+  }
+  public sealed class VulkanMemoryAllocator : IDisposable
+  {
+    public Device Device { get; }
+    public VulkanMemoryAllocator(in VulkanMemoryAllocatorCreateInfo createInfo);
+    public int CurrentFrameIndex { get; set; }
+    public void Dispose();
+    public ref readonly Silk.NET.Vulkan.PhysicalDeviceProperties PhysicalDeviceProperties { get; }
+    public ref readonly PhysicalDeviceMemoryProperties MemoryProperties { get; }
+    public MemoryPropertyFlags GetMemoryTypeProperties(int memoryTypeIndex);
+    public int? FindMemoryTypeIndex(uint memoryTypeBits, in AllocationCreateInfo allocInfo);
+    public int? FindMemoryTypeIndexForBufferInfo(
+      in BufferCreateInfo bufferInfo,
+      in AllocationCreateInfo allocInfo);
+    public int? FindMemoryTypeIndexForImageInfo(
+      in ImageCreateInfo imageInfo,
+      in AllocationCreateInfo allocInfo);
+    public Allocation AllocateMemory(
+      in MemoryRequirements requirements,
+      in AllocationCreateInfo createInfo);
+    public Allocation AllocateMemoryForBuffer(
+      Buffer buffer,
+      in AllocationCreateInfo createInfo,
+      bool BindToBuffer = false);
+    public Allocation AllocateMemoryForImage(
+      Image image,
+      in AllocationCreateInfo createInfo,
+      bool BindToImage = false);
+    public Result CheckCorruption(uint memoryTypeBits);
```

Comment:
Is corruption detection enabled by default ?

Reply:
I haven't implemented it myself, but it was part of VMA, and we can try to provide it as a debug tool type thing.

Reply:
Well, we could for the time being remove it from the public facing api part and add it later.